typings.d.ts

```diff
@@ -1,139 +1,247 @@
      */
-    minify?: boolean | {},
-    cache: boolean,
+    minify?: boolean | MinifyOptions;
     /**
      * Render errors into the HTML page
      */
-    showErrors: boolean,
+    showErrors?: boolean;
     /**
-     * List all entries which should be injected
+     * The `webpack` require path to the template.
+     * @see https://github.com/jantimon/html-webpack-plugin/blob/master/docs/template-option.md
      */
-    chunks: 'all' | string[],
+    template?: string;
     /**
-     * List all entries which should not be injeccted
+     * Allow to use a html string instead of reading from a file
      */
-    excludeChunks: string[],
-    chunksSortMode: 'auto' | 'manual' | (((entryNameA: string, entryNameB: string) => number)),
+    templateContent?:
+      | false // Use the template option instead to load a file
+      | string
+      | Promise<string>;
     /**
-     * Inject meta tags
+     * Allows to overwrite the parameters used in the template
      */
-    meta: false // Disable injection
-      | {
-          [name: string]: string|false // name content pair e.g. {viewport: 'width=device-width, initial-scale=1, shrink-to-fit=no'}`
-          | {[attributeName: string]: string|boolean} // custom properties e.g. { name:"viewport" content:"width=500, initial-scale=1" }
-      },
+    templateParameters?:
+      | false // Pass an empty object to the template function
+      | ((
+          compilation: any,
+          assets,
+          assetTags: {
+            headTags: HtmlTagObject[];
+            bodyTags: HtmlTagObject[];
+          },
+          options: Options
+        ) => { [option: string]: any })
+      | ((
+          compilation: any,
+          assets,
+          assetTags: {
+            headTags: HtmlTagObject[];
+            bodyTags: HtmlTagObject[];
+          },
+          options: Options
+        ) => Promise<{ [option: string]: any }>)
+      | { [option: string]: any };
+    /**
+     * The title to use for the generated HTML document
+     */
+    title?: string;
     /**
      * Enforce self closing tags e.g. <link />
      */
-    xhtml: boolean
-
+    xhtml?: boolean;
     /**
      * In addition to the options actually used by this plugin, you can use this hash to pass arbitrary data through
      * to your template.
      */
     [option: string]: any;
-}
+  }
 
-/**
- * A tag element according to the htmlWebpackPlugin object notation
- */
-interface HtmlTagObject {
   /**
-   * Attributes of the html tag
-   * E.g. `{'disabled': true, 'value': 'demo'}`
+   * The internally used options. This is a variant of the Options interface but with the types
+   * set up to match the expectations in index.js
    */
-  attributes: {
-    [attributeName: string]: string|boolean
-  },
-  /**
-   * Wether this html must not contain innerHTML
-   * @see https://www.w3.org/TR/html5/syntax.html#void-elements
-   */
-  voidTag: boolean,
+  interface InternalOptions extends Required<Omit<Options, "minify">> {
```

Comment:
Could you please add a short comment explaining what you are doing here and especially why minify is not required?  
This might save the next one some time :)

Reply:
Sure. `Required` and `Omit` already have comments regarding their origins further up the file but I can add some additional context.